src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -1162,6 +1167,41 @@ void InterpreterSystemQuery::waitLoadingParts()
             "Command WAIT LOADING PARTS is supported only for MergeTree table, but got: {}", table->getName());
     }
 }
+void InterpreterSystemQuery::loadPrimaryKeys()
+{
+    if (!table_id.empty())
+    {
+        getContext()->checkAccess(AccessType::SYSTEM_LOAD_PRIMARY_KEY, table_id.database_name, table_id.table_name);
+        StoragePtr table = DatabaseCatalog::instance().getTable(table_id, getContext());
+
+        if (auto * merge_tree = dynamic_cast<MergeTreeData *>(table.get()))
+        {
+            LOG_TRACE(log, "Loading primary keys for table {}", table_id.getFullTableName());
+            merge_tree->loadPrimaryKeys();
+        }
+        else
+        {
+            throw Exception(ErrorCodes::BAD_ARGUMENTS,
+                "Command LOAD PRIMARY KEY is supported only for MergeTree table, but got: {}", table->getName());
+        }
+    }
+    else
+    {
+        getContext()->checkAccess(AccessType::SYSTEM_LOAD_PRIMARY_KEY);
+        LOG_TRACE(log, "Loading primary keys for all tables");
+
+        for (auto & database : DatabaseCatalog::instance().getDatabases())
+        {
+            for (auto it = database.second->getTablesIterator(getContext()); it->isValid(); it->next())
+            {
+                if (auto * merge_tree = dynamic_cast<MergeTreeData *>(it->table().get()))
+                {
+                    merge_tree->loadPrimaryKeys();
```

Comment:
It will be better if this process will be multithreaded. Please take a look at `IO/SharedThreadPools.h` and check out how they are used in other codebase.

Reply:
Sure, can you please also take a look at this ticket even though it is not relevant to code changes here? #66251 
Since it is quite vital for doing unit tests, really appreciate it!

Reply:
Thank you for your response, I have make changes on the logic using multithreaded method you mentioned above!